testsuite/smoke-test/src/consensus/consensus_fault_tolerance.rs

```diff
@@ -213,6 +213,54 @@ async fn test_no_failures() {
     .unwrap();
 }
 
+#[tokio::test]
+async fn test_faulty_votes() {
```

Comment:
how did you verify this works as expected? we don't have any logs showing pessimistic validators being added?

Reply:
I have added some `println!` states in validator_verifier and checked that the authors that are added to pessimistic verify set are being verified pessimistically from next time.
<img width="1507" alt="Screenshot 2024-10-04 at 1 37 19 PM" src="https://github.com/user-attachments/assets/cb725dca-2597-4aec-8bb2-ef96dbf57a10">
<img width="1496" alt="Screenshot 2024-10-04 at 1 36 22 PM" src="https://github.com/user-attachments/assets/4b72a8c6-c3e5-47a9-9670-7940b39cbdbb">